webapp/src/Controller/Jury/ContestController.php

```diff
@@ -504,8 +508,61 @@ public function editAction(Request $request, int $contestId)
                 $problem->setContest($contest);
             }
 
+            // Determine the removed teams, team categories and problems.
+            // Note that we do not send out create / update events for
+            // existing / new problems, teams and team categories. This happens
+            // when someone connects to the event feed (or we have a
+            // dependent event) anyway and adding the code here would
+            // overcomplicate this function
```

Comment:
It's not clear to me what this code does exactly and why. Is this already generating a partial cascade of delete events?

Reply:
A no, it checks what teams (/categories/problems) *used* to be set on the contest but are not anymore. It would be wayyyy better to use array_diff here, but that only works on strings and not on objects, because logic.

So the *only* thing I want is that if I remove a team(/category/problem) from a contest, it emits a delete event (since we dont pick these up any other way).